### PR TITLE
Some more compat annotations

### PR DESCRIPTION
**Pull request by fredrikekre**
Compat annotation for `unique!(f, A)`, #30141.
Compat annotation for `rank(A; rtol=..., atol=...)`, #29926.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
I don't know what the convention is, if any, for where to put the annotation within docstrings, but I would have voted for the end, after the examples. That said, you've done a bunch of these already, so I fully trust your judgement here. 😉

**Comment by fredrikekre** [comment]
> I don't know what the convention is, if any, for where to put the annotation within docstrings, but I would have voted for the end, after the examples.

Imagine how disappointed you would be after reading about an amazing function and even looked how useful it is from the examples, just to see that you have the wrong Julia version!

File: stdlib/LinearAlgebra/src/generic.jl [hunk 1]
```diff
@@ -713,7 +713,7 @@ end
 
 """
     rank(A::AbstractMatrix; atol::Real=0, rtol::Real=atol>0 ? 0 : n*ϵ)
-    rank(A::AbstractMatrix, rtol::Real) = rank(A; rtol=rtol)  # to be deprecated in Julia 2.0
+    rank(A::AbstractMatrix, rtol::Real)
```
[the remark this answers is not in the document]
Comment (fredrikekre) [reply stdlib/LinearAlgebra/src/generic.jl]: Yes, I included that information in the note, and we have not used this style anywhere else.

File: stdlib/LinearAlgebra/src/generic.jl [hunk 2]
```diff
@@ -722,6 +722,11 @@ tolerances, respectively. The default relative tolerance is `n*ϵ`, where `n`
 is the size of the smallest dimension of `A`, and `ϵ` is the [`eps`](@ref) of
 the element type of `A`.
 
+!!! compat "Julia 1.1"
+    The `atol` and `rtol` keyword arguments requires at least Julia 1.1.
+    In Julia 1.0 `rtol` is available as a positional argument, but this
+    will be deprecated in Julia 2.0.
```
Comment (innerlee) [inline stdlib/LinearAlgebra/src/generic.jl]: "also available"

Comment (fredrikekre) [reply stdlib/LinearAlgebra/src/generic.jl]: No, on 1.0 it is not available as a kwarg.

File: stdlib/LinearAlgebra/src/generic.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (StefanKarpinski) [reply stdlib/LinearAlgebra/src/generic.jl]: Agree, we should only document the methods and keywords that we actually want people to use. Any other supported-for-compatibility but no-longer-recommended methods/keywords should be in notes like this.

**Pull request merged**